modules/tests/shared/src/test/scala/io/circe/JsonObjectSuite.scala

```diff
@@ -2,200 +2,237 @@ package io.circe
-    val result2 = JsonObject.fromFoldable(fields)
-    val expected = fields.find(_._1 == key).nonEmpty
+  property("contains should find fields if they exist") {
+    forAll { (fields: List[(String, Json)], key: String) =>
+      val result1 = JsonObject.fromIterable(fields)
+      val result2 = JsonObject.fromFoldable(fields)
+      val expected = fields.find(_._1 == key).nonEmpty
 
-    assert(result1.contains(key) === expected)
-    assert(result2.contains(key) === expected)
+      assertEquals(result1.contains(key), expected)
+      assertEquals(result2.contains(key), expected)
+    }
   }
 
-  "size" should "return the expected result" in { (fields: List[(String, Json)]) =>
-    val result1 = JsonObject.fromIterable(fields)
-    val result2 = JsonObject.fromFoldable(fields)
-    val expected = fields.toMap.size
+  property("size should return the expected result") {
+    forAll { (fields: List[(String, Json)]) =>
+      val result1 = JsonObject.fromIterable(fields)
+      val result2 = JsonObject.fromFoldable(fields)
+      val expected = fields.toMap.size
 
-    assert(result1.size === expected)
-    assert(result2.size === expected)
+      assertEquals(result1.size, expected)
+      assertEquals(result2.size, expected)
+    }
   }
 
-  "isEmpty" should "return the expected result" in { (fields: List[(String, Json)]) =>
-    val result1 = JsonObject.fromIterable(fields)
-    val result2 = JsonObject.fromFoldable(fields)
-    val expected = fields.isEmpty
+  property("isEmpty should return the expected result") {
+    forAll { (fields: List[(String, Json)]) =>
+      val result1 = JsonObject.fromIterable(fields)
+      val result2 = JsonObject.fromFoldable(fields)
+      val expected = fields.isEmpty
 
-    assert(result1.isEmpty === expected)
-    assert(result2.isEmpty === expected)
+      assertEquals(result1.isEmpty, expected)
+      assertEquals(result2.isEmpty, expected)
+    }
   }
 
-  "nonEmpty" should "return the expected result" in { (fields: List[(String, Json)]) =>
-    val result1 = JsonObject.fromIterable(fields)
-    val result2 = JsonObject.fromFoldable(fields)
-    val expected = fields.nonEmpty
+  property("nonEmpty should return the expected result") {
+    forAll { (fields: List[(String, Json)]) =>
+      val result1 = JsonObject.fromIterable(fields)
+      val result2 = JsonObject.fromFoldable(fields)
+      val expected = fields.nonEmpty
 
-    assert(result1.nonEmpty === expected)
-    assert(result2.nonEmpty === expected)
+      assertEquals(result1.nonEmpty, expected)
+      assertEquals(result2.nonEmpty, expected)
+    }
   }
 
-  "kleisli" should "find fields if they exist" in { (fields: List[(String, Json)], key: String) =>
-    val result1 = JsonObject.fromIterable(fields)
-    val result2 = JsonObject.fromFoldable(fields)
-    val expected = fields.find(_._1 == key).map(_._2)
+  property("kleisli should find fields if they exist") {
+    forAll { (fields: List[(String, Json)], key: String) =>
```

Comment:
@zarthross This will have the same issues as the one you fixed where the generator isn't generating a ton of 'good' test cases.  Do you think just using that one here as well would be good?  One thing with that generator is that it doesn't create cases where it wouldn't exist at all, if we expect that base case to be here as well.

Reply:
> @zarthross This will have the same issues as the one you fixed where the generator isn't generating a ton of 'good' test cases. Do you think just using that one here as well would be good?

Yes, I think that's a good idea to use a similar generator.

> One thing with that generator is that it doesn't look like it will create cases where the key wouldn't exist at all in the list of (key,vals) since it's always appended onto the list, if we expect that base case to be here as well.

You maybe right... The generator I wrote might just be putting in `JNull` instead of it simply not existing.  It shouldn't be hard to fix that generator so that sometimes its `Null` and othertimes its completely missing.  I think that would make for a better test.

Reply:
35b30d7 adds in the tests to cover those edge cases and pulls out the generators for use in the tests.